### PR TITLE
Feature/query microservices

### PR DESCRIPTION
**Pull request by jwomeara**

### CODE REVIEW

**Review by jwomeara — COMMENTED** [review]
Notable changes in this PR:
 - Major version and dependency bumps
 - Query Metric class package changes
 - ShardTableQueryMetricHandler (Remote & Local), technically added in spring boot 2.x, which allow for query via rest or locally run.

File: service/src/test/java/datawave/microservice/querymetric/QueryMetricTestBase.java [hunk 1]
```diff
@@ -534,7 +534,7 @@ protected void ensureDataWritten(Cache incomingCache, Cache lastWrittenCache, St
         long now = System.currentTimeMillis();
         Config config = ((HazelcastCacheManager) this.cacheManager).getHazelcastInstance().getConfig();
         MapStoreConfig mapStoreConfig = config.getMapConfig(incomingCache.getName()).getMapStoreConfig();
-        int writeDelaySeconds = Math.min(mapStoreConfig.getWriteDelaySeconds(), 1000);
+        int writeDelaySeconds = 2 * Math.min(mapStoreConfig.getWriteDelaySeconds(), 1000);
```
Comment (apmoriarty) [inline service/src/test/java/datawave/microservice/querymetric/QueryMetricTestBase.java:537]: Was this test flakey without the 2x multiplier?

**Review by apmoriarty — APPROVED** [review]

File: service/src/test/java/datawave/microservice/querymetric/QueryMetricTestBase.java (hunk 1 above)
Comment (alerman) [reply service/src/test/java/datawave/microservice/querymetric/QueryMetricTestBase.java:537]: In general, i don't love how this writeDelay is being handled. we are setting it to a config value, then doubling it, then adding a second below on line 539. I am also curious if the test was flaky.

The variable naming is also suspect. This is not a delay but a timeout.

**Review by alerman — COMMENTED** [review]
Looks good, just want to understand the test changes better before I can approve

File: service/src/test/java/datawave/microservice/querymetric/QueryMetricTestBase.java (hunk 1 above)
Comment (jwomeara) [reply service/src/test/java/datawave/microservice/querymetric/QueryMetricTestBase.java:537]: i think this 2x multiplier just needs to be cleaned up.  over the 3 years of developing this effort, there were times when running all of the tests in a multi-threaded fashion for the build caused some timeout-related tests to fail.  I don't remember exactly, but this is bringing back some vague feelings of frustration trying to get things to work.  I'll remove it and see if it's still necessary.

Comment (jwomeara) [reply service/src/test/java/datawave/microservice/querymetric/QueryMetricTestBase.java]: everything seems good without the 2x modifier, so i'll mark this as resolved.

**Review by matthpeterson — APPROVED** [review]

**Pull request merged**